twiggy/__init__.py

```diff
@@ -79,6 +84,47 @@ def quickSetup(*args, **kwargs):
     return quick_setup(*args, **kwargs)
 
 
+def dict_config(config):
+    """
+    Configure twiggy logging via a dictionary
+
+    :arg config: a dictionary which configures twiggy's outputs and emitters.  See
+        :attr:`TWIGGY_CONFIG_SCHEMA` for details of the format of the dict.
+
+    .. seealso:: :ref:`dict_config` for a thorough explaination of the outputs and emitters
+        concepts from the dictionary
+
+    .. versionadded: 0.5
+    """
+    try:
+        cfg = _validate_config(config)
+    except ValueError as e:
```

Comment:
Hmm  I think in this case it makes sense to raise... There is no validation for programmatic setup (since there's no clear 'done' state at which to check) so it falls back to internal errors at runtime, but that would be really surprising behavior here IMO